lib/features/pokemon_details/pokemon_details_page.dart

```diff
@@ -0,0 +1,98 @@
+import 'package:flutter/material.dart';
+import 'package:pokedexbootcamp/api/model/pokemon_detail.dart';
+import 'package:pokedexbootcamp/utils/async.dart';
+import 'package:pokedexbootcamp/utils/constants.dart';
+import 'package:pokedexbootcamp/utils/theme.dart';
+import 'package:pokedexbootcamp/widgets/image_widget_pokemon_details.dart';
+import 'package:pokedexbootcamp/widgets/row_description.dart';
+
+class PokemonDetailsPage extends StatefulWidget {
+  const PokemonDetailsPage({
+    Key? key,
+    required this.pokemonDetail,
+    required this.pokemonName,
+  }) : super(key: key);
+
+  final Async<PokemonDetail?> pokemonDetail;
+  final String pokemonName;
+
+  @override
+  State<StatefulWidget> createState() => _PokemonDetailsPageState();
+}
+
+class _PokemonDetailsPageState extends State<PokemonDetailsPage> {
+  @override
+  void dispose() {
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: colorWhite,
+      appBar: AppBar(
+        leading: IconButton(
+          icon: const Icon(Icons.arrow_back, color: Colors.white),
+          onPressed: () => Navigator.of(context).pop(),
+        ),
+        title: const Text(
+          pokemonAppBarTitle,
+          style: TextStyle(
+            fontSize: fontSizeAppBar,
+          ),
+        ),
+        centerTitle: true,
+      ),
+      body: widget.pokemonDetail.when(
+        error: (errorMessage) {
+          WidgetsBinding.instance.addPostFrameCallback((_) => _showErrorMessageSnackbar(context, errorMessage));
+          return const Center(
+            child: Text(pokemonOfflineMessageLabel),
+          );
+        },
+        (data) => Center(
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            crossAxisAlignment: CrossAxisAlignment.center,
+            children: [
+              Center(
+                child: Text(
+                  widget.pokemonName,
+                  style: const TextStyle(
+                    fontSize: fontSizePokemonDetailName,
+                    fontWeight: FontWeight.bold,
+                  ),
+                ),
+              ),
+              Row(
+                mainAxisAlignment: MainAxisAlignment.center,
+                crossAxisAlignment: CrossAxisAlignment.center,
+                children: [
+                  PokemonDetailsImages(pokeImage: data!.sprite!.frontDefault!),
```

Comment:
why create data property nullable when in the end you're just gonna end of enforcing them to become non-null 🤔